src/libstd/io/util.rs

```diff
@@ -28,6 +28,22 @@ use io::{self, Read, Write, ErrorKind, BufRead};
 /// This function will return an error immediately if any call to `read` or
 /// `write` returns an error. All instances of `ErrorKind::Interrupted` are
 /// handled by this function and the underlying operation is retried.
+///
+/// # Examples
+///
+/// ```
+/// use std::io;
+///
+/// # fn foo() -> io::Result<()> {
+/// let mut reader = &(b"hello")[..];
+/// let mut writer: Vec<u8> = vec![];
+///
+/// try!(io::copy(&mut reader, &mut writer));
+///
+/// assert_eq!(&reader[..], &writer[..]);
```

Comment:
There's quite a bit of syntax going on here! I feel like this assert shouldn't need the slicing syntax, and the `&(b"...")[...]` above may want to be replaced with type ascription:

```rust
let mut reader: &[u8] = b"hello";
```